### DISCUSSION TITLE
Adding records on behalf of another user

### DISCUSSION THREAD

**Discussion by lxnunes** [Budibase/budibase#3726] [Help] [answered]
Hi! This is my 1st post and I have only been acquainted with this amazing tool since a few hours. I ran into a few issues partly due to lack of exploring and partly due to lack of caffeine... So I apologise in advance if this has been answered already or is just dumb obvious.

I am trying to setup an application where an Admin would add user records and activities on their behalf so that later the user may access the app and consult his/her records. The default Users table is related to the Persons table and there is a 2nd relationship between Persons and Activities (one person may have many Activities)

My 1st question: How could I restrict the records returned to the user?  I have seen that when setting up the DataProvider I can set the Data to the Relationship "Current User.Persons-Created By". I believe this would solve my problem if I could somehow transfer the "created By" from the Admin to the User's id (is it an id?). Is there a way to achive this? FYI I have tried also adding a column with the user email to all tables, and use it as a filter for the records presented to the user but it seems a bit of a blunt solution.

My 2nd question: In practice the Users and Persons table have a one-to-one relationship, I just need to populate an email column in Persons and add extra fields such as phone number, birth date, etc. Is there anything against using the default Users table to store all the data I need about my users? I noticed the Add Screen button does not seem to foresee to automatically generate screens and I am a bit puzzled...

Many thanks for your time, I hope I was clear enough :)

**Comment by mjashanks** [comment, marked as answer]
Hi @lxnunes - welcome!

The best way to do this is to just use your `Users` table - i.e. no `Persons` table. You can add all the columns that you need to the `Users` table, and this data will only apply to users in this app.

The only drawback of this method is that you cannot have a `Person` who is not a `User`. Everyone needs a user account.

If this is not an issue, then this guide should explain what you need to do to display activities for the current user: https://docs.budibase.com/guides/items-assigned-to-me

Let me know if this is a problem, and I'll try to build your original request!

**Reply by lxnunes** [reply]
Hi @mjashanks, thank you for replying back! 
Indeed all users will have an account so that guide should work.
Again, thank you for the help.
Cheers
